torch/fx/experimental/merge_matmul.py

```diff
@@ -9,20 +9,27 @@
 from typing import Dict, List
 
 
-def get_first_dim(t: torch.Tensor) -> int:
+def split_result_tensors(result: torch.Tensor, inputs: List[torch.Tensor]) -> List[torch.Tensor]:
     """
-    A free function primarily for use in the merge_matmul graph transformation below
-    that returns the first dimension of a Tensor. This is necessary because torch.Tensor.shape
-    is an attribute (and cannot be the target of a call_function node) and also helps save
-    a getitem op in the graph.
+    A free function for use in the merge_matmul graph transformation below that
+    splits the output from a merged matmul into the individual results for each
+    input tensor.
 
     Arguments:
-        t: The tensor to get the first dimension of.
+        result: The merged matmul result tensor.
+        inputs: The list of inputs that were merged into one for the matmul.
 
     Returns:
-        The first dimension of t.
+        List of matmul results for each input tensor.
     """
-    return t.shape[0]
+    splits = [x.shape[0] for x in inputs]
+
+    # When fx tracer is running, x.shape[0] will be torch.fx.Attribute but we
+    # need an int even when tracing
+    if isinstance(result, torch.fx.Proxy):
+        splits = [0] * len(inputs)
```

Comment:
Tiny nit, marginally inefficient when tracing 🙂 
```suggestion
    # When fx tracer is running, x.shape[0] will be torch.fx.Attribute but we
    # need an int even when tracing
    if isinstance(result, torch.fx.Proxy):
        splits = [0] * len(inputs)
    else:
        splits = [x.shape[0] for x in inputs]
```